python/essex-config/essex_config/config.py

```diff
@@ -0,0 +1,134 @@
+"""Main configuration module for the essex_config package."""
+
+from functools import cache
+from typing import Any, ClassVar, TypeVar
+
+from pydantic import BaseModel
+from pydantic_core import PydanticUndefined
+
+from essex_config.configuration_field import ConfigurationField, FieldType
+from essex_config.sources import EnvSource, Source
+
+DEFAULT_SOURCE_LIST: list[Source] = [EnvSource()]
+
+T = TypeVar("T", bound="Config")
+
+
+class Config(BaseModel):
+    """Base class for configuration."""
+
+    sources: ClassVar[list[Source]] = DEFAULT_SOURCE_LIST
+
+    @staticmethod
+    def __resolve_value(name: str, data: dict[str, Any]):
```

Comment:
Since this is intended to be used as much as possible, I wonder if we can up the docs, and ideally include examples?
Would also be a nice example on how we should document :)

WDYT? Not a blocker for this PR.

Reply:
The docs for sources are in the readme and there are examples in the class docs :)